plugins/example_plugin/mod.ts

```diff
@@ -45,4 +46,12 @@ export class ExamplePlugin extends Plugin<ExamplePluginConfig> {
       }
     }
   }
+
+  // deno-lint-ignore no-explicit-any
+  configSchema(): z.ZodObject<any> | undefined {
```

Comment:
What do you think about injecting `z` as a `configSchema` argument? I would even rename it to something else (`s` from schema?), so that if, in the future, we need to pivot from `zod` we can do it without much headaches.

The intention is not to wrap or abstract `zod` completely, but make it easer to replace if necessary, but no hard feelings.

Reply:
> What do you think about injecting `z` as a `configSchema` argument? I would even rename it to something else (`s` from schema?), so that if, in the future, we need to pivot from `zod` we can do it without much headaches.
> 
> The intention is not to wrap or abstract `zod` completely, but make it easer to replace if necessary, but no hard feelings.

I like the intent but not sure how providing `z` as a arg would make it easier if it's not abstracted, are you able to provide a bit more detail here please?

I'll merge this PR for now and follow-up if we decide to go with providing `z` as a arg